pkg/task/taskid/taskid.go

```diff
@@ -27,69 +27,86 @@ import (
 // TaskReferenceId, but "foo.bar#qux" cannot be used as a TaskReferenceId because it
 // has a suffix.
 
-type TaskReferenceId struct {
+// TODO: rewrite comments above.
+
+type UntypedTaskReference interface {
```

Comment:
Could you resolve the TODO in this PR? (if any blockers exist pleas let me know)
I think the difference `UntypedTaskReference` and `TaskReference` is very difficult for most contributors who want to develop a new feature.

Reply:
Added comments.